### PR TITLE
refactor: onload listener for stylesheets in before hook of tests

### PR DESCRIPTION
**Pull request by jkodu**
This PR refactors the `addStyleSheet` helper for CSSOM based tests into `axe.testUtils` object. The function also returns a promise, thereby allowing to wait until the stylesheets are loaded in the `before` hook, thus ensuring assets are available before running respective tests.

Closes issue: 
- https://github.com/dequelabs/axe-core/issues/1086

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [X] Follows the commit message policy, appropriate for next version (MS: needs squashing)
- [X] Has documentation updated, a DU ticket, or requires no documentation change
- [X] Includes new tests, or was unnecessary
- [X] Code is reviewed for security by: << Marcy Sutton >>

### CODE REVIEW

File: test/integration/full/preload-cssom/preload-cssom.js [hunk 1]
```diff
@@ -44,14 +27,15 @@ describe('preload cssom integration test', function() {
...
-			// wait for network request to complete for added sheets
-			setTimeout(done, 5000);
+			axe.testUtils
+				.addStyleSheets(styleSheets)
+				.then(function() {
+					done();
+				})
+				.catch(function(error) {
+					console.error('Could not load stylesheets for testing.', error);
+					done();
```
Comment (WilcoFiers) [inline test/integration/full/preload-cssom/preload-cssom.js]: Seems like this should cause the tests to fail.

File: test/integration/full/preload/preload.js [hunk 2]
```diff
@@ -82,9 +63,16 @@ describe('preload integration test', function() {
...
-			setTimeout(done, 5000);
+			// load stylesheets
+			axe.testUtils
+				.addStyleSheets(styleSheets)
+				.then(function() {
+					done();
+				})
+				.catch(function(error) {
+					console.error('Could not load stylesheets for testing.', error);
+					done();
```
Comment (WilcoFiers) [inline test/integration/full/preload/preload.js]: Same here, if injecting fails, the before hook should fail.

File: test/testutils.js [hunk 3]
```diff
@@ -220,4 +222,64 @@ testUtils.awaitNestedLoad = function awaitNestedLoad(win, cb) {
...
+ */
+testUtils.addStyleSheets = function addStyleSheets(sheets) {
+	var promises = [];
+
+	sheets.forEach(function(data) {
+		var p = axe.testUtils.addStyleSheet(data);
+		promises.push(p);
+	});
+
+	return Promise.all(promises);
```
Comment (WilcoFiers) [inline test/testutils.js]: This won't work in IE.

**Comment by jkodu** [comment]
@WilcoFiers 

Usage of `Promise` has been removed in these test updated. Please review again.

**Pull request merged**